File: ports/csharp/src/core/io/BcsDir.cs
using System.IO;

namespace BCS.Core.IO {
    // BcsDir abstracts standard directory operations for C#
    public class BcsDir {
        private string _path;

        public BcsDir(string path) {
            _path = path;
        }

        public bool Exists() {
            return Directory.Exists(_path);
        }

        public void Mkdir() {
            Directory.CreateDirectory(_path); // CreateDirectory handles full paths implicitly in C#
        }

        public void MkdirPath() {
            Directory.CreateDirectory(_path);
        }
    }
}

File: ports/java/src/core/io/BcsDir.java
package bcs.core.io;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.nio.file.Path;

// BcsDir abstracts standard directory operations for Java
public class BcsDir {
    private final Path path;

    public BcsDir(String pathStr) {
        this.path = Paths.get(pathStr);
    }

    public boolean exists() {
        return Files.isDirectory(this.path);
    }

    public void mkdir() throws IOException {
        Files.createDirectory(this.path);
    }

    public void mkdirPath() throws IOException {
        Files.createDirectories(this.path);
    }
}
